Storybook prop addons

webb04 opened the pull request:
## Why are you doing this?

We can easily share the components we develop across the team.
Gets deployed to: https://guardian.github.io/apps-rendering

## Changes

- Add `knobs` addon
- Deploy to GitHub pages script
- GitHub action to deploy changes daily
- Use fonts inside of public folder

JamieB-gu commented on .storybook/preview-head.html:
```diff
@@ -0,0 +1,22 @@
+    .js-email-sub__iframe{width:100%;}
+    .js-email-sub__iframe + figcaption{margin-top:-8px;}
+    video, .element-atom iframe, .element-audio iframe{width:100%;}
+</style>
```
Does this differ from the styles in `page.tsx`?

webb04 replied:
Nope, but I couldn't think of a better way of including these global styles in the storybook components...

JamieB-gu commented on .storybook/preview-head.html:
```diff
@@ -0,0 +1,22 @@
+    figure{margin:1em 0;}
+    figure.element-embed{margin:2em 0;}
+    .js-email-sub__iframe{width:100%;}
+    .js-email-sub__iframe + figcaption{margin-top:-8px;}
```
We have email functionality?

webb04 replied:
Yep the signup forms within articles. I'm not sure if they are interactives or atoms but they're probably not showing on apps-rendering yet.

webb04 replied:
These styles should probably be moved into `ArticleBody`

JamieB-gu commented on package.json:
```diff
@@ -17,7 +17,8 @@
     "build:server": "webpack --config webpack.config.js --config-name server",
     "build:client": "webpack --config webpack.config.js --config-name client",
-    "storybook": "start-storybook"
+    "storybook": "start-storybook -s ./public",
```
What does `-s ./public` do? Does it put output in the repo `/public` directory or in a public directory inside `.storybook`?

webb04 replied:
I've changed this and now uploaded the fonts to s3.

This would work running storybook locally but didn't work on GitHub pages (loading fonts)

alexduf commented:
As we chatted about, I think it's a good idea to expose the storybook on github pages, however two problems remaining to solve:
 - hosting the fonts directly with github page
 - finding a way for a github action to have the right to push to the github page

webb04 commented:
- Added a shell script to push the fonts to `gh-pages`
- Deleted the GitHub action for the time being as more work needs to be done to add credentials
